tfsdk/value_from_test.go

```diff
@@ -0,0 +1,164 @@
+package tfsdk
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+type person struct {
+	Name    types.String `tfsdk:"name"`
+	Age     types.Int64  `tfsdk:"age"`
+	OptedIn types.Bool   `tfsdk:"opted_in"`
+	Address types.List   `tfsdk:"address"`
```

Comment:
Is it worth testing other types (e.g., `types.Map`), along the lines of
```go
	Address types.Map    `tfsdk:"address"`
```

Then using the following in `personAttrTypes`:
```go
		"address": types.MapType{
			ElemType: types.StringType,
		},
```

Populating `mrX` with:
```go
		Address: types.Map{
			ElemType: types.StringType,
			Elems: map[string]attr.Value{
				"number": types.String{Value: "1"},
				"street": types.String{Value: "Beckford Close"},
				"town":   types.String{Value: "Gotham"},
			},
		},
```

And finally testing `expectedMrXObj`:
```go
			"address": types.Map{
				ElemType: types.StringType,
				Elems: map[string]attr.Value{
					"number": types.String{Value: "1"},
					"street": types.String{Value: "Beckford Close"},
					"town":   types.String{Value: "Gotham"},
				},
			},
```

Reply:
Just added it ;)